shell/shared/grain.js

```diff
@@ -386,6 +389,57 @@ Meteor.methods({
       return outerResult;
     }
   },
+  requestAccess: function (origin, grainId, identityId) {
+    check(origin, String);
+    check(grainId, String);
+    check(identityId, String);
+    if (!this.isSimulation) {
+      const grain = Grains.findOne(grainId);
+      if (!grain) {
+        throw new Meteor.Error(404, "No such grain");
+      }
+      const grainOwner = globalDb.getUser(grain.userId);
+      const email = _.findWhere(SandstormDb.getUserEmails(grainOwner), {primary: true});
+      if (!email) {
+        throw new Meteor.Error("no email", "Grain owner has no email address.");
+      }
+      const emailAddress = email.email;
+
+      const identity = globalDb.getIdentity(identityId);
+      globalDb.addContact(grainOwner._id, identityId);
+
+      const fromEmail = "Sandstorm server <no-reply@" + HOSTNAME + ">";
+      const senderEmails = SandstormDb.getVerifiedEmails(identity);
+      const replyTo = senderEmails.length > 0 ? senderEmails[0] : fromEmail;
+
+      // TODO(soon): In the HTML version, we should display an identity card.
+      let identityNote = "";
+      if (identity.profile.service === "google") {
+        identityNote = " (" + identity.privateIntrinsicName + " on Google)";
+      } else if (identity.profile.service === "github") {
+        identityNote = " (" + identity.profile.intrinsicName + " on GitHub)";
+      } else if (contact.profile.service === "email") {
+        identityNote = " (" + identity.profile.intrinsicName + ")";
+      }
+
+      const message = identity.profile.name + identityNote +
```

Comment:
I was going to edit the `From` field in the email to also include the display name, but this comment dissuaded me.

I like the message text as-is, but I'd be receptive to thoughts from @zarvox.